### PR TITLE
Add power Euclidean distance and mean

### PR DESCRIPTION
**Pull request by qbarthelemy**
@xpennec, can you confirm that, for p=0, power Euclidean mean tends to log-Euclidean mean (rather than AIR mean)? Thx.

### CODE REVIEW

File: pyriemann/utils/distance.py [hunk 1]
```diff
@@ -323,6 +323,65 @@ def distance_logeuclid(A, B, squared=False):
...
 
+def distance_poweuclid(A, B, p, squared=False):
+    r"""Power Euclidean distance between SPD/HPD matrices.
+
+    The power Euclidean distance of order :math:`p` between two SPD/HPD
+    matrices :math:`\mathbf{A}` and :math:`\mathbf{B}` is [1]_:
+
+    .. math::
+        d(\mathbf{A},\mathbf{B}) = \frac{1}{|p|}
+        \Vert \mathbf{A}^p - \mathbf{B}^p \Vert_F
```
Comment (VascoSch92) [inline pyriemann/utils/distance.py:334]: What is F in this case? The vector space?

Comment (qbarthelemy) [reply pyriemann/utils/distance.py:334]: F for Frobenius norm.

File: pyriemann/utils/distance.py [hunk 2]
```diff
@@ -323,6 +323,65 @@ def distance_logeuclid(A, B, squared=False):
...
+
+    References
+    ----------
+    .. [1] `Power Euclidean metrics for covariance matrices with application to
+        diffusion tensor imaging
+        <https://arxiv.org/abs/1009.3045>`_
+        I.L. Dryden, X. Pennec, & J.M. Peyrat. arXiv, 2010.
+    """
+    if not isinstance(p, (int, float)):
+        raise ValueError("Exponent p must be a scalar exponent")
```
Comment (VascoSch92) [inline pyriemann/utils/distance.py:369]:
```suggestion
        raise ValueError(f"Exponent p must be a scalar, but got {p}")
```

**Review by gcattan — APPROVED** [review]
All good for me. I understand you had the confirmation you wanted from [@xpennec](https://github.com/xpennec)

File: pyriemann/utils/distance.py [hunk 3]
```diff
@@ -323,6 +323,65 @@ def distance_logeuclid(A, B, squared=False):
...
+
+    References
+    ----------
+    .. [1] `Power Euclidean metrics for covariance matrices with application to
+        diffusion tensor imaging
+        <https://arxiv.org/abs/1009.3045>`_
+        I.L. Dryden, X. Pennec, & J.M. Peyrat. arXiv, 2010.
+    """
+    if not isinstance(p, (int, float)):
+        raise ValueError("Exponent p must be a scalar (Got {type(p)})")
```
Comment (VascoSch92) [inline pyriemann/utils/distance.py:369]: This is an f-string right? You forgot the f :)

File: pyriemann/utils/distance.py [hunk 4]
```diff
@@ -323,6 +323,65 @@ def distance_logeuclid(A, B, squared=False):
...
+
+    if p == 1:
```
Comment (VascoSch92) [inline pyriemann/utils/distance.py:371]: Nit: I will put the if-block in increasing order of p. So it is easier ti read

File: pyriemann/utils/distance.py (hunk 3 above)
Comment (qbarthelemy) [reply pyriemann/utils/distance.py:369]: Yes, you are right!

**Pull request merged**